### ISSUE TITLE
OwnTracks Bad Request no log entries

### ISSUE THREAD

**Issue by pantherale0** [traccar/traccar#5486]
**Describe the bug**
Using OwnTracks as a tracker and Caddy as a reverse proxy (and providing SSL) for the HTTP server running at port 5144, it seems the server returns a HTTP 400 status code, however there are no log entries that show that actually help resolving the issue

**To Reproduce**
1. Setup latest traccar in docker container
2. Setup Caddy server using following config for http app
```
"traccar-owntracks": {
          "listen": [":5144"],
          "routes": [
            {
              "handle": [
                {
                  "handler": "reverse_proxy",
                  "transport": {
                    "protocol": "http"
                  },
                  "upstreams": [
                    {
                      "dial": "127.0.0.1:5144"
                    }
                  ]
                }
              ],
              "match": [
                {
                  "host": [
                    "traccar.example.com"
                  ]
                }
              ],
              "terminal": true
            }
          ]
        }
```
3. Configure OwnTracks in HTTP mode:
`Endpoint = https://traccar.example.com`
`Device ID = 1234`
`Tracker ID = 1`
`Username = example`
4. Create device in Traccar with identifier `owntracks/example/1234`
5. Wait and check logs

**Expected behavior**
An entry in the log file that states why a 400 status code is being raised, but this doesn't happen.

**Screenshots**
N/A, log below
```
2024-12-31 10:24:49  INFO: [Ta2e08722: owntracks < 172.16.55.205] POST / HTTP/1.1\r\nHost: traccar.example.com:5144\r\nUser-Agent: Owntracks-Android/gms/420503003\r\nContent-Length: 127\r\nAccept-Encoding: gzip\r\nContent-Type: application/json; charset=utf-8\r\nX-Forwarded-For: 140.228.92.32\r\nX-Forwarded-Host: traccar.example.com:5144\r\nX-Forwarded-Proto: https\r\nX-Limit-D: 1234\r\nX-Limit-U: jordan\r\n\r\n{"_type":"status","_id":"6491e098","android":{"hib":1,"bo":0,"loc":0,"ps":1,"wifi":1},"topic":"owntracks/example/1234/status"}
2024-12-31 10:24:49  INFO: [Ta2e08722: owntracks > 172.16.55.205] HTTP/1.1 400 Bad Request\r\ncontent-length: 0\r\n\r\n
```

**Smartphone (please complete the following information):**
 - Device: [e.g. iPhone6] S23 Ultra
 - OS: [e.g. iOS8.1] Android 14
 - Browser [e.g. stock browser, safari] OwnTracks
 - Version [e.g. 22] 2.5.3

**Additional context**
I wonder if this is actually because the lat / long is missing in the payload? It would be useful for Traccar to either silently discard these messages (returning a 200 status code) as OwnTracks will queue these messages and keep retrying them constantly without a timeout as far as I can see.

**Comment by tananaev** [comment]
Yeah, I'm pretty sure it's because the location is missing.

**Comment by ckrey** [comment]
You are sending a "status" message, not a "location" message from the OwnTracks app. 
Are you using an older version of the OwnTracks app? see owntracks/android#1873

**Comment by pantherale0** [comment]
Nope, using latest version from play store, (2.5.3). The behaviour reported in that issue is something that I don't experience (the app itself does not hang like reported in that issue).

**Comment by tananaev** [comment]
Do you have the full protocol documentation for the format?

**Comment by ckrey** [comment]
Full description is here: https://owntracks.org/booklet/tech/json/

**Comment by tananaev** [comment]
This should fix the problem: https://github.com/traccar/traccar/commit/0f0bb36283bba13657d666220f3cb0d06cf44859.

**Comment by ckrey** [comment]
@thanks for fixing the "status" message problem.

But the removal of lines 60...69 causes exiting integrations to fail. "tid" is only a 1-letter abbreviation of the full "topic" 
and is only used as a fallback if "topic" is missing.

<img width="479" alt="Bildschirmfoto 2025-01-05 um 12 41 23" src="https://github.com/user-attachments/assets/caa9ca5c-2e21-44ba-8acb-7b033a32edfe" />

**Comment by tananaev** [comment]
Do you have a sample that's failing?

**Comment by ckrey** [comment]
Device is created in traccar with `identifier = owntracks/ckrey/22A67880-15C3-41A6-9157-25545C7993AC`

Test with original payload from OwnTracks iOS App:
```
curl http://127.0.0.1:5144 -d '{"lon":-122,"acc":5,"created_at":1736161415,"lat":37,"topic":"owntracks/ckrey/22A67880-15C3-41A6-9157-25545C7993AC","t":"u","m":1,"tst":1736161169,"conn":"w","alt":0,"_type":"location","tid":"AC"}'
```

```
2025-01-06 12:19:07  INFO: [T7fa5ec14] connected
2025-01-06 12:19:07  INFO: [T7fa5ec14: owntracks < 127.0.0.1] POST / HTTP/1.1\r\nHost: 127.0.0.1:5144\r\nUser-Agent: curl/8.7.1\r\nAccept: */*\r\nContent-Length: 196\r\nContent-Type: application/x-www-form-urlencoded\r\n\r\n{"lon":-122,"acc":5,"created_at":1736161415,"lat":37,"topic":"owntracks/ckrey/22A67880-15C3-41A6-9157-25545C7993AC","t":"u","m":1,"tst":1736161169,"conn":"w","alt":0,"_type":"location","tid":"AC"}
2025-01-06 12:19:07  WARN: Unknown device - AC (127.0.0.1)
2025-01-06 12:19:07  INFO: [T7fa5ec14: owntracks > 127.0.0.1] HTTP/1.1 400 Bad Request\r\ncontent-length: 0\r\n\r\n
2025-01-06 12:19:07  INFO: [T7fa5ec14] disconnected
```

**Comment by tananaev** [comment]
So what is the case for using TID then?

**Comment by ckrey** [comment]
The `tid` defaults to the las two letters of the deviceid, which is used to construct the topic, But it can be set independently to e.g. the owner's initials:
![Screenshot 2025-01-08 at 08 54 30](https://github.com/user-attachments/assets/f8fa506a-82d9-437d-9321-8ec291c38ed3)
[screenshot from here](https://owntracks.org/booklet/features/tid/)

**Comment by tananaev** [comment]
Then it sounds like we should always be using only topic and never tid. Is that correct?

**Comment by ckrey** [comment]
yes, use the topic as traccar's  uniqueId

**Comment by mbirth** [comment]
Any chance to get the Docker images updated with the fix?
